doc/sphinx/language/core/basic.rst

```diff
@@ -499,6 +502,18 @@ The following attribute is supported by every command:
 
    Alias of :attr:`warnings`.
 
+Document-level attributes
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. cmd:: Library Attributes {* #[ {*, @attribute } ] }
+   :name: Library Attributes
+
+   This command associates attributes to the whole current
+   document. When compiled with ``coqc`` (see Section
+   :ref:`_thecoqcommands`), the attributes are associated to the
+   compiled file and may have an effect when the file is loaded with
+   :cmd:`Require`.
```

Comment:
```suggestion
   Associates attributes with the
   document. When compiled with ``coqc`` (see Section
   :ref:`_thecoqcommands`), the attributes are associated with the
   compiled file and may have an effect when the file is loaded with
   :cmd:`Require`.
```

I'm not sure the "When ..." adds much.  It will depend on the attribute.  

FWIW, I think the documentation uses `file` and `document` almost interchangeably but isn't explicit about their relationship.  For coqc, it is definitely a file.  In CoqIDE, for example, you can work in a buffer without saving it as a file.